x-pack/plugins/cloud_security_posture/public/pages/vulnerability_dashboard/vulnerability_table_panel_section.tsx

```diff
@@ -159,11 +159,17 @@ export const VulnerabilityTablePanelSection = () => {
               defaultMessage: 'CVSS',
             }
           ),
-          render: (cvss: { score: number; version: string }) => (
-            <EuiLink onClick={() => onCellClick({ 'vulnerability.score.base': cvss.score })}>
-              <CVSScoreBadge version={cvss.version} score={cvss.score} />
-            </EuiLink>
-          ),
+          render: (cvss: PatchableVulnerabilityStat['cvss']) => {
+            if (!cvss.score || !cvss.version) {
+              return null;
+            }
+
+            return (
+              <EuiLink onClick={() => onCellClick({ 'vulnerability.score.base': cvss.score! })}>
```

Comment:
```suggestion
              <EuiLink onClick={() => onCellClick({ 'vulnerability.score.base': cvss.score })}>
```

i might be missing something but this shouldn't be needed if you already verified it exist

Reply:
for some reason, TS is not picking up the early return in case `cvss.score` is missing and I couldn't figure out why quickly. Do you have an idea why it might be?

Reply:
I tried it on my side and i loled. looks like a typescript bug. consulted with @orouz the master and those two solutions work:

```typescript
render: (cvss: PatchableVulnerabilityStat['cvss']) => {
            if (!cvss.score || !cvss.version) {
              return null;
            }
            const score = cvss.score;
            // const query = { 'vulnerability.score.base': cvss.score };

            return (
              <EuiLink onClick={() => onCellClick({ 'vulnerability.score.base': score})}>
              // <EuiLink onClick={() => onCellClick(query)}>
                <CVSScoreBadge version={cvss.version} score={cvss.score} />
              </EuiLink>
            );
          },
```

you can either declare a new constant for `score`, or the entire `query` object on a new constant. either way the inferring seems bugged inside the onClick function.